tests/integration/test_headers_and_subtotals.py

```diff
@@ -3810,3 +3811,540 @@ def it_provide_residual_test_stats_including_hs(self):
+            ]
+            - (
+                slice_.counts[
+                    4,
+                ]
+                + slice_.counts[
+                    5,
+                ]
+            ),
+        )
+        # --- Subdiff in col 0 equals (col=1 + col=2) - (col=5) ---
+        np.testing.assert_array_equal(
+            slice_.counts[:, 0],
+            (slice_.counts[:, 1] + slice_.counts[:, 2]) - (slice_.counts[:, 5]),
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_col_margin(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+        columns_margin = slice_.columns_margin
+        # --- Subdiff in col 0 equals (col=1 + col=2) - (col=5) ---
+        np.testing.assert_almost_equal(
+            columns_margin[0],
+            (columns_margin[1] + columns_margin[2]) - (columns_margin[5]),
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_row_margin(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+        rows_margin = slice_.rows_margin
+        # --- Subdiff in row 0 equals row=1 - (col=4 + col=5) ---
+        np.testing.assert_almost_equal(
+            rows_margin[0], rows_margin[1] - (rows_margin[4] + rows_margin[5])
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_col_bases(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+        column_bases = slice_.column_weighted_bases
+        # --- Bases are ALWAYS additive (even with differences) ---
+        # --- Subdiff in col 0 equals (col=1 + col=2) + (col=5) ---
+        np.testing.assert_almost_equal(
+            column_bases[:, 0],
+            column_bases[:, 1] + column_bases[:, 2] + column_bases[:, 5],
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_row_bases(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+        row_bases = slice_.row_weighted_bases
+        # --- Bases are ALWAYS additive (even with differences) ---
+        # --- Subdiff in row 0 equals row=1 + (row=4 + row=5) ---
+        np.testing.assert_almost_equal(
+            row_bases[
+                0,
+            ],
+            row_bases[
+                1,
+            ]
+            + row_bases[
+                4,
+            ]
+            + row_bases[
+                5,
+            ],
+        )
+
+    def test_cat_subdiff_x_cat_subdiff_table_prop(self, row_col_subdiff_insertions):
+        slice_ = Cube(
+            CR.CAT_SUBDIFF_X_CAT_SUBDIFF, transforms=row_col_subdiff_insertions
+        ).partitions[0]
+
+        np.testing.assert_almost_equal(
+            slice_.table_proportions, slice_.counts / slice_.table_base
```

Comment:
I notice these `/` aren't python2 compatible. Advice on what to change to?
https://github.com/Crunch-io/crunch-cube/pull/253/checks?check_run_id=1942716810#step:5:142

Reply:
fixed via `from __future__ import division`